### PR TITLE
Query the full consensus parameters in the pool and solo miners

### PR DESCRIPTION
**Pull request by danield9tqh**
## Summary
Add the consensus parameters to pool and solo miner to be able to switch at the FishHash activation sequence.

## Testing Plan
Unit test + running pool miner / solo miner on devnet

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish/src/serde/BlockTemplateSerde.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
-import { Block } from '../primitives/block'
+import { Block, RawBlock } from '../primitives/block'
```
Comment (mat-if) [inline ironfish/src/serde/BlockTemplateSerde.ts:5]: Are these changes intended to be in this PR?

File: ironfish/src/rpc/clients/client.ts [hunk 2]
```diff
@@ -733,6 +734,24 @@ export abstract class RpcClient {
       ).waitForEnd()
     },
 
+    getParsedConsensusParameters: (
```
Comment (mat-if) [inline ironfish/src/rpc/clients/client.ts:737]: I'm not sure this is the best approach

1. This is duplicating code of the `getConsensusParameters`
2. `number | null` is a better type, especially for consumers of the SDK (modifying ConsensusParameters itself is a topic for another day..)
3. This is a new pattern that deviates from the expectations of the existing pattern - RPC client functions return RPC responses - leading to an inconsistent API for consumers of the SDK

If we do need this, my initial feeling is that this should be a util function

Comment (danield9tqh) [reply ironfish/src/rpc/clients/client.ts:737]: Ok yea I can just duplicate it in the 2 places we call it. I was back and forth on this

File: ironfish/src/serde/BlockTemplateSerde.ts (hunk 1 above)
Comment (danield9tqh) [reply ironfish/src/serde/BlockTemplateSerde.ts:5]: They are. The hashing code deals with `RawBlock` and `RawBlock` header a lot so its very helpful to be able to go from `SerializedBlockTemplate` to `RawBlock` and back. I can do it in another PR though

**Review by mat-if — APPROVED** [review]
